Add information about changes in Java PSI (#318)
JetBrains/intellij-sdk-docs: reference_guide/api_changes/api_changes_list_2020.md

## 2020.3

### Changes in IntelliJ Platform 2020.3

> `com.intellij.psi.PsiTypeElement`: PSI structure of multi-dimensional arrays in Java source files changed.

Reviewer: same here
> NOTE: Entries not starting with code quotes (`name`) can be added to document non-code changes and will be skipped in API verification.

> Now the children are flattened: brackets for all the dimensions are direct children of the `PsiTypeElement` that represent the multi-dimensional array. This change doesn't break source or binary compatibility but may produce behavioral change in the code that traverses the tree of Java source files.

Reviewer 2: ```suggestion
Now the children are flattened: brackets for all the dimensions are direct children of the `PsiTypeElement` that represent the multi-dimensional array. This change doesn't break source or binary compatibility but may produce behavioral changes in the code that traverses the tree of Java source files.
```

> `com.intellij.psi.PsiAnnotation.getOwner` method now returns `PsiType` instead of `PsiTypeElement` for type annotations in Java source files.

Reviewer: please follow rule
> NOTE: Entries not starting with code quotes (`name`) can be added to document non-code changes and will be skipped in API verification.

> This allows to identify where type annotation belongs to if it's attached to an inner class or a particular dimension of multi-dimensional array. This change doesn't break source or binary compatibility but may produce behavioral changes for callers.

Reviewer 2: ```suggestion
This change supports identifying whether a type annotation is attached to an inner class or a particular dimension of a multi-dimensional array.
This change doesn't break source or binary compatibility but may produce behavioral changes for callers.
```

### Changes in IntelliJ Platform 2020.3

The PSI structure of multi-dimensional arrays in Java source files changed (see `com.intellij.psi.PsiTypeElement`).

> Now the children are flattened: brackets for all the dimensions are direct children of the `PsiTypeElement` that represent the multi-dimensional array. This change doesn't break source or binary compatibility but may produce behavioral changes in the code that traverses the tree of Java source files.

Reviewer: sorry missed, description must start with `:` (same below)

Author: Probably it's a good idea to automate these checks to avoid spending time on human review rounds?
